### PR TITLE
Refactor official build to use arcade templates

### PR DESCRIPTION
**Pull request by ericstj**
This refactors the official build to use the arcade templates which automate a lot of validation tasks.  It also removes some redundant signing we were doing before, opting for a single signing phase after all packages have been built.

Here's a successful build from this change https://dev.azure.com/dnceng/internal/_build/results?buildId=2030382&view=results

### CODE REVIEW

File: src/Microsoft.ML.Mkl.Redist/Microsoft.ML.Mkl.Redist.csproj [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <Import Project="$(RepoRoot)eng/pkg/Pack.props" />
   <PropertyGroup>
-    <Authors>Intel</Authors>
```
Comment (tarekgh) [inline src/Microsoft.ML.Mkl.Redist/Microsoft.ML.Mkl.Redist.csproj:4]: ><Authors>Intel</Authors> [](http://example.com/codeflow?start=4&length=24)

I assume this is ok and no concern removing it.

File: src/Native/Native.proj [hunk 2]
```diff
@@ -143,8 +143,10 @@
   </Target>
 
   <Target Name="Pack" />
-  <Target Name="Restore" />
   <Target Name="Test" />
 
+  <ItemGroup>
+    <PackageDownload Include="MlNetMklDeps" Version="[$(MlNetMklDepsVersion)]" />
```
Comment (tarekgh) [inline src/Native/Native.proj:149]: >[ [](http://example.com/codeflow?start=53&length=1)

interesting, does the version included inside brackets?

**Comment by tarekgh** [comment]
@LittleLittleCloud Any idea about this failure? 

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-machinelearning-refs-pull-6412-merge-b739effdcc32498c89/Microsoft.ML.AutoML.Tests/1/console.8ab99b89.log?helixlogtype=result
```
 ---> System.ObjectDisposedException: The CancellationTokenSource has been disposed.
   at System.Threading.CancellationTokenSource.Cancel()
   at Microsoft.ML.AutoML.AutoMLExperiment.<>c__DisplayClass26_1.<RunAsync>g__handler|4(Object o, EventArgs e) in /__w/1/s/src/Microsoft.ML.AutoML/AutoMLExperiment/AutoMLExperiment.cs:line 270
   at Microsoft.ML.AutoML.AggregateTrainingStopManager.<.ctor>b__4_0(Object o, EventArgs e) in /__w/1/s/src/Microsoft.ML.AutoML/AutoMLExperiment/IStopTrainingManager.cs:line 129
   at Microsoft.ML.AutoML.TimeoutTrainingStopManager.<.ctor>b__5_0(Object o, EventArgs e) in /__w/1/s/src/Microsoft.ML.AutoML/AutoMLExperiment/IStopTrainingManager.cs:line 72
   at Microsoft.ML.AutoML.CancellationTokenStopTrainingManager.<.ctor>b__5_0() in /__w/1/s/src/Microsoft.ML.AutoML/AutoMLExperiment/IStopTrainingManager.cs:line 38
   at System.Threading.CancellationToken.<>c.<Register>b__12_0(Object obj)
   at System.Threading.CancellationTokenSource.CallbackNode.<>c.<ExecuteCallback>b__9_0(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
```

**Review by tarekgh — APPROVED** [review]
To my best knowledge, LGTM.

File: src/Native/Native.proj (hunk 2 above)
Comment (ericstj) [reply src/Native/Native.proj:149]: Yes, PackageDownload requires explicit version references. https://learn.microsoft.com/en-us/nuget/consume-packages/packagedownload-functionality

File: src/Microsoft.ML.Mkl.Redist/Microsoft.ML.Mkl.Redist.csproj (hunk 1 above)
Comment (ericstj) [reply src/Microsoft.ML.Mkl.Redist/Microsoft.ML.Mkl.Redist.csproj:4]: It was wrong for us to claim Intel is the Author of the package.  The gallery doesn't even present it to our users.
https://www.nuget.org/packages/Microsoft.ML.Mkl.Redist/2.0.0-preview.22313.1
image
We are the ones who build this package and binary, it's a [custom linked](https://github.com/dotnet/machinelearning/blob/04dda55ab0902982b16309c8e151f13a53e9366d/docs/building/MlNetMklDeps/README.md) copy of MKL.  The package itself has the Intel license.

**Review by michaelgsharp — APPROVED** [review]
:shipit:

**Pull request merged**